runtime/makelib/targets.mk.osx.inc.ftl

```diff
@@ -118,7 +118,7 @@ endif
 </#if>
 
 # https://stackoverflow.com/questions/21907504/how-to-compile-shared-lib-with-clang-on-osx
-UMA_DLL_LINK_FLAGS += -shared -undefined dynamic_lookup -install_name @rpath/lib$(UMA_TARGET_NAME).dylib
+UMA_DLL_LINK_FLAGS += -shared -undefined dynamic_lookup -install_name @rpath/lib$(UMA_LIB_NAME).dylib
```

Comment:
Since this is an ftl file, can we do with this `$(UMA_TARGET_NAME)${uma.buildinfo.version.major}${uma.buildinfo.version.minor}`?  

I'm loath to add new defines to UMA as we're trying to replace it with cmake

Reply:
Adding `$major$minor` to every dylib causes this error: `JVMJ9VM013W Initialization error in function j9dyn29(7): failed to initialize JImage interface`.

Reply:
**com.ibm.j9.uma.platform.PlatformImplementation.getTargetNameWithRelease:**
```
	public String getTargetNameWithRelease(Artifact artifact) {
		if ( artifact.appendRelease() ) {
			return artifact.getTargetName() + getStream();
		}
		return artifact.getTargetName();
	}
```

**module.xml**: `<artifact name="jvm" ... appendrelease="false">`

If `appendrelease` is true, then `$major$minor` is appended.

Reply:
Is `appendrelease="false"` enabled on the wrong libraries?  While the PR addresses the issue, it feels like it leaves the larger problem lurking.

Reply:
> Is appendrelease="false" enabled on the wrong libraries?

I don't think so. `appendrelease="false"` is set to false for libraries which don't change across Java versions such as `jvm`, `jsig`, `j9omrport` etc. Libraries such as `j9vm`, `j9gc`, `j9prt` ... may differ across Java versions. For such libraries, `appendrelease` is not specified. I think `appendrelease` is set to true by default if it's not specified.

Reply:
Thanks @babsingh for digging through this with me deskside.  I now agree that this is a reasonable solution given the existing hacks in j9lib.h.in / j9lib.h.ftl.

FYI @dnakamura This may affect the cmake build when ported to osx as it adds a new UMA flag `UMA_LIB_NAME`